src/config/options.js

```diff
@@ -101,6 +101,61 @@ const rootSection = section(
                 "": { files: [] }, // Use `all` set with default values if sets were not specified in a config
             },
         ),
+
+        devServer: section({
+            command: options.optionalString("devServer.command"),
+            env: options.optionalObject("devServer.env"),
+            args: options.optionalArray("devServer.args"),
+            cwd: options.optionalString("devServer.cwd"),
+            logs: options.optionalBoolean("devServer.logs"),
+            readinessProbe: option({
+                defaultValue: defaults.devServer.readinessProbe,
+                validate: value => {
+                    if (typeof value === "function" || value === null) {
+                        return;
+                    }
+
+                    if (!_.isPlainObject(value)) {
+                        throw new Error("devServer.readinessProbe must be a function, object or null");
+                    }
+
+                    if (!_.isUndefined(value.url) && typeof value.url !== "string" && value.url !== null) {
+                        throw new Error("devServer.readinessProbe.url must be a string or null");
+                    }
+
+                    if (
+                        !_.isUndefined(value.isReady) &&
+                        typeof value.isReady !== "function" &&
+                        value.isReady !== null
+                    ) {
+                        throw new Error("devServer.readinessProbe.isReady must be a function or null");
+                    }
+
+                    if (!_.isUndefined(value.timeouts) && !_.isPlainObject(value.timeouts)) {
+                        throw new Error("devServer.readinessProbe.timeouts must be an object");
+                    }
+
+                    if (value.timeouts) {
+                        ["waitServerTimeout", "probeRequestTimeout", "probeRequestInterval"].forEach(name => {
+                            if (!_.isUndefined(value.timeouts[name]) && typeof value.timeouts[name] !== "number") {
+                                throw new Error(`devServer.readinessProbe.timeouts.${name} must be a number`);
+                            }
+                        });
+                    }
+                },
+                map: value =>
+                    typeof value === "function"
+                        ? value
+                        : {
+                              ...defaults.devServer.readinessProbe,
+                              ...(value || {}),
+                              timeouts: {
+                                  ...defaults.devServer.readinessProbe.timeouts,
+                                  ...((value && value.timeouts) || {}),
+                              },
+                          },
```

Comment:
applying deep defaults, so user can specify one timeout config and the rest would be used from default settings